app/src/org/commcare/activities/HomeScreenBaseActivity.java

```diff
@@ -335,6 +345,15 @@ private boolean doLoginLaunchChecksInOrder() {
             return true;
         }
 
+        // In case a sync request from FCM was made while the user was logged out, this will
+        // trigger a blocking sync
+        if (HiddenPreferences.isPendingSyncRequestFromServer() &&
+                HiddenPreferences.getPendingSyncRequestFromServerTime()>SyncDetailCalculations.getLastSyncTime()) {
```

Comment:
This is for when the user is logged out when the sync request comes in, should we just trigger the sync (in the next login) without checking the request time against the last sync?

Reply:
Any successful sync though should reset `isPendingSyncRequestFromServer` to false, that will make the time comparison insignificant (since if the sync happened after the firebase request, `isPendingSyncRequestFromServer` would still be `false`)